test/hotspot/jtreg/compiler/c2/TestSerialAdditions.java

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright (c) 2024 Red Hat and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.c2;
+
+import compiler.lib.ir_framework.*;
+import compiler.lib.ir_framework.Test;
+
+/*
+ * @test
+ * @bug 8325495
+ * @summary C2 should optimize for series of Add of unique value. e.g., a + a + ... + a => a*n
+ * @library /test/lib /
+ * @run driver compiler.c2.TestSerialAdditions
+ */
+public class TestSerialAdditions {
+    public static void main(String[] args) {
+        TestFramework.run();
+    }
+
+    @DontInline
+    private static void verifyResult(int base, int factor, int observed) {
+        int expected = base * factor; // compute expected result here while making sure not inlined in callers
+        if (expected != observed) {
+            throw new AssertionError("Expected " + expected + " but got " + observed);
+        }
+    }
+
+    @DontInline
+    private static void verifyResult(long base, long factor, long observed) {
+        long expected = base * factor;
+        if (expected != observed) {
+            throw new AssertionError("Expected " + expected + " but got " + observed);
+        }
+    }
+
+    // ----- integer tests -----
+    @Test
+    @Arguments(values = {Argument.RANDOM_EACH})
+    @IR(counts = { IRNode.ADD_I, "1" })
+    @IR(failOn = {IRNode.LSHIFT_I})
```

Comment:
Generally, for single strings, you can remove the braces:
```suggestion
    @IR(failOn = IRNode.LSHIFT_I)
```